### ISSUE TITLE
island netherportals aren't linked correctly to the skyworld_nether world

### ISSUE THREAD

**Issue by wesley27** [rlf/uSkyBlock#856]
Hello again! Just reinstalled uSkyBlock on a fresh new server. It created skyworld and skyworld_nether. I'm able to teleport to skyworld_nether via multiverse commands, and it is indeed a nether world. However, if I enter a nether portal while on my island in skyworld, it creates a new map "skymap_nether," that is a normal overworld map. It doesn't tell multiverse about it either- by this I mean when I do /mvinfo while in "skymap_nether" multiverse says this world exists but multiverse doesn't know about it. Every time I go through a portal in skyworld, it takes me to this world.

I haven't looked at any of the code, I've just been testing everything in the plugin as I work to re-build my skyblock server. I would assume that the last time you changed something in uSkyBlock where portals and the maps are managed, you forgot to change the map name from skymap_nether to skyworld_nether. Could be wrong, again I haven't looked at your code, but that would make sense.

Thanks guys :)

_Please paste the output from `/usb version` below_
```
Console	: [19:36:15 INFO]: Name: uSkyBlock
Console	: [19:36:15 INFO]: Version: 2.7.2
Console	: [19:36:15 INFO]: Description: Ultimate SkyBlock v2.7.2-45f756-430
Console	: [19:36:15 INFO]: Language: en (en)
Console	: [19:36:15 INFO]:   State: d=128, r=64, i=1, p=1, n=true, awe=false
Console	: [19:36:15 INFO]: Server: CraftBukkit git-Spigot-90f61bc-5e5cf84 (MC: 1.10.2)
Console	: [19:36:15 INFO]:   State: online=false, bungee=true
Console	: [19:36:15 INFO]: ------------------------------
Console	: [19:36:15 INFO]: Vault 1.5.6-b49 (ENABLED)
Console	: [19:36:15 INFO]: WorldEdit 6.1.3;7a097ca (ENABLED)
Console	: [19:36:15 INFO]: WorldGuard 6.1.2;e38d98d (ENABLED)
Console	: [19:36:15 INFO]: Multiverse-Core 2.5-b717 (ENABLED)
Console	: [19:36:15 INFO]: ------------------------------
```

_What steps will reproduce the problem?_
1. Install uSkyBlock on a server. It will create the worlds "skymap" and "skymap_nether"
2. Create an island, and build a nether portal on your island (this will be in world "skymap")
3. Enter the netherportal. It will create a new normal world called "skyworld_nether" and using your nether portal will only take you to this world

**Comment by wesley27** [comment]
Also, I just noticed that the "skyworld_nether" generated by loading the plugin is a pretty much empty nether map with a lava floor and a netherrack ceiling...

**Comment by RhysRS** [comment]
Have you tried linking nether portals in skyworld and skyworld_nether using multiverse?

**Comment by wesley27** [comment]
Yes, that's exactly what I did. Only, it wasn't as simple as setting up multiverse-netherportals. I had to delete the crazy-generated (as described above) "skyworld-nether" world generated by uSkyBlock, delete the "skymap_nether" normal map generated by entering a netherportal in the "skyworld" world, and finally I had to manually create a "skyworld_nether" nether world via multiverse commands. Then, I was able to link them with multiverse-netherportals.

TL;DR - Yes, that's how I got it working. However, nowhere in this plugin's documentation does it say it requires multiverse-netherportals to properly link the worlds together. And it shouldn't, something was probably missed when updating the world and portal managing code. However, as long as mvnp is required, you should probably add it to your documentation.

**Comment by rlf** [comment]
It's not required, but it will use MVNP to setup the worlds for you - if it's available.

If you don't have it, you need to point the portals your self (using the portal plugin of your choice).

**Comment by dutchy1001** [comment]
Wesley27 said:
> However, nowhere in this plugin's documentation does it say it requires multiverse-netherportals to properly link the worlds together

This is documented on the wiki [here](https://github.com/rlf/uSkyBlock/wiki/Installation) as **_Required dependencies_** and on our bukkit page [here](https://dev.bukkit.org/bukkit-plugins/uskyblock/pages/installation/) as **_Highly recommended dependencies are (needed for the nether versions)_**.

I know their "softdepended" plugins and not needed for **IF** you don't want the nether or our nether feature to work but they are mentioned to be needed for the nether.

Wesley27 said:
> Also, I just noticed that the "skyworld_nether" generated by loading the plugin is a pretty much empty nether map with a lava floor and a netherrack ceiling...

This is part of the nether feature and is meant this way, you should have an island here simular to the "overworld" island but only made of netherrack and a few other blocks including the portal. When breaking the blocks new blocks should appear, this is called _terraforming_.

**Comment by wesley27** [comment]
Ah, I didn't look at the wiki, it isn't mentioned at all on the Spigot page [here.](https://www.spigotmc.org/resources/uskyblock.2280/)

Thanks for the clarification on the nether, I was not aware of that new feature. Do all nether mobs still spawn normally? Ghasts and pigmen? How do you find blazes and wither skeletons without netherfortresses?

**Comment by rlf** [comment]
https://github.com/rlf/uSkyBlock/wiki/Features#nether
